lock-api/src/main/java/com/palantir/lock/client/LockRefreshingTimelockService.java

```diff
@@ -92,23 +96,36 @@ public LockResponse lock(LockRequest request) {
 
     @Override
     public WaitForLocksResponse waitForLocks(WaitForLocksRequest request) {
-        return delegate.waitForLocks(request);
+        return executeOnTimeLock(() -> delegate.waitForLocks(request));
     }
 
     @Override
     public Set<LockToken> refreshLockLeases(Set<LockToken> tokens) {
-        return delegate.refreshLockLeases(tokens);
+        return executeOnTimeLock(() -> delegate.refreshLockLeases(tokens));
     }
 
     @Override
     public Set<LockToken> unlock(Set<LockToken> tokens) {
         lockRefresher.unregisterLocks(tokens);
-        return delegate.unlock(tokens);
+        return executeOnTimeLock(() -> delegate.unlock(tokens));
     }
 
     @Override
     public long currentTimeMillis() {
-        return delegate.currentTimeMillis();
+        return executeOnTimeLock(delegate::currentTimeMillis);
+    }
+
+    private <T> T executeOnTimeLock(Callable<T> callable) {
+        try {
+            return callable.call();
+        } catch (Exception e) {
+            if (e.getCause() instanceof ConnectException
+                    || e.getCause() instanceof UnknownHostException) {
+                throw Throwables.unwrapAndThrowAtlasDbDependencyException(e);
+            } else {
+                throw Throwables.throwUncheckedException(e);
```

Comment:
Doesnt this mean that not all exceptions are `AtlasDbDependencyException`. 
Also, the delegate impls might be wrapping the exception into `PalantirRuntimeExceptions` etc.. So we can end up with chains of the kind `AtlasDbDependencyException(PalantirRuntimeException(ConnecException))`.
I would prefer if instead of catching and wrapping the exception(or the cause of an exception if its ExecutionException etc.) where it originated.

Reply:
Not all exceptions should be treated as `AtlasDbDependencyException`. For example, a request to fetch 0 timestamps could be issued, and this would return an `IllegalArgumentException`.

Reply:
I was thinking more along the lines that this case will be an `AtlasDbDependencyException` with cause being the `IllegalArgumentException`. We should discuss about what will be more valuable to the application in rubix.

Reply:
I still think its better to consistently throw AtlasDbDependencyException, but that can be a separate PR too, when we are testing in the k8s environment.